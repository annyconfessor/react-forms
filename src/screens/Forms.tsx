import React from "react";

import Input from "../components/Input/Input.tsx";

import { Container, Wrapper, Box, WrapperButtons, TextWrapper, ResetButton, SendButton } from "./styles.tsx";

import { mockDataInfos } from "./mocks.ts";

const Forms = () => {
  return (
    <>
    <Container>
      <Box>
          {mockDataInfos.map((item) => (<><Wrapper key={item.id}><TextWrapper>{item.field}</TextWrapper><Input /></Wrapper></>))}
        <WrapperButtons>
          <ResetButton>Limpar</ResetButton>
          <SendButton>Enviar</SendButton>
        </WrapperButtons>
      </Box>
    </Container>
    </>
  );
}

export default Forms;
