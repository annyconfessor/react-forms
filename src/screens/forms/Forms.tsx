import React, { useState } from "react";

import Input from "../../components/Input/Input.tsx";
import Button from "../../components/Button/Button.tsx";
import Modal from "../modal/Modal.tsx";

import { Container, Wrapper, Box, WrapperButtons, TextWrapper } from "./styles.tsx";

import { mockDataInfos } from "./mocks.ts";

const Forms = () => {
const [modal, setModal] = useState(false);
const [value, setValue] = useState(``);

  const handleModal = () => {
    setModal(true);
  }

  const handleValue = (e) => {
    const event = e.target.value
    setValue(event)
  }

  return (
    <>
    <Container>
      <Box>
          {mockDataInfos.map((item) => (
            <>
              <Wrapper key={item.id}>
                <TextWrapper>{item.field}</TextWrapper>
                  <Input 
                    type="text"
                    name="textInput"
                    value={value} 
                    onchange={handleValue}
                  />
              </Wrapper>
            </>
          ))}
        <WrapperButtons>
          <Button variant="reset">Limpar</Button>
          <Button variant="send" href="abrir-modal" onClick={handleModal}>Enviar</Button>
        </WrapperButtons>
      </Box>
      {modal && <Modal />}
    </Container>
    </>
  );
}

export default Forms;
