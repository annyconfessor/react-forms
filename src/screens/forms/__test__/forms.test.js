import renderer from 'react-test-renderer'
import Forms from '../Forms'

it('should render corectly', () => {
  const matchRender = renderer.create(<Forms />).toJSON()
  expect(matchRender).toMatchSnapshot()
})