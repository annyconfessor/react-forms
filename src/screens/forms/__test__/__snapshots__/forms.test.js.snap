// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render corectly 1`] = `
<div
  className="sc-jSMfEi jVlKoF"
>
  <div
    className="sc-gKXOVf lawOGm"
  >
    <div
      className="sc-ftvSup jdIqOh"
    >
      <div
        className="sc-iBkjds ljzrWh"
      >
        Name
      </div>
      <input
        className="sc-bczRLJ"
        onChange={[Function]}
      />
    </div>
    <div
      className="sc-ftvSup jdIqOh"
    >
      <div
        className="sc-iBkjds ljzrWh"
      >
        Sobrenome
      </div>
      <input
        className="sc-bczRLJ"
        onChange={[Function]}
      />
    </div>
    <div
      className="sc-ftvSup jdIqOh"
    >
      <div
        className="sc-iBkjds ljzrWh"
      >
        E-mail
      </div>
      <input
        className="sc-bczRLJ"
        onChange={[Function]}
      />
    </div>
    <div
      className="sc-ftvSup jdIqOh"
    >
      <div
        className="sc-iBkjds ljzrWh"
      >
        Estado
      </div>
      <input
        className="sc-bczRLJ"
        onChange={[Function]}
      />
    </div>
    <div
      className="sc-ftvSup jdIqOh"
    >
      <div
        className="sc-iBkjds ljzrWh"
      >
        Observacao
      </div>
      <input
        className="sc-bczRLJ"
        onChange={[Function]}
      />
    </div>
    <div
      className="sc-papXJ kyCLiB"
    >
      <button
        className="sc-gsnTZi jXDvbx"
      >
        Limpar
      </button>
      <button
        className="sc-gsnTZi kUqLNt"
        href="abrir-modal"
        onClick={[Function]}
      >
        Enviar
      </button>
    </div>
  </div>
</div>
`;
